favor.lic

```diff
@@ -2,17 +2,88 @@
   Documentation: https://elanthipedia.play.net/Lich_script_repository#favor
 =end
 
-custom_require.call(%w[common common-travel equipmanager])
+custom_require.call(%w[common common-travel common-money common-items equipmanager])
 
 class CrossingFavor
   include DRC
   include DRCT
+  include DRCM
+  include DRCI
 
   def initialize
+    arg_definitions = [
+      [
+        { name: 'immortal', regex: /\w+/, optional: true, description: "Immortal who's orb to get from an altar, turns on altar favors instead of the puzzle" }
+      ]
+    ]
+
+    args = parse_args(arg_definitions, true)
+    @settings = get_settings
+    @hometown = @settings.hometown
+    if args.immortal
+      @immortal = args.immortal.capitalize
+      @all_theurgy_data = get_data('theurgy')
+      @data = @all_theurgy_data[@hometown]
+      @immortal_to_aspect = @all_theurgy_data['immortal_to_aspect']
+      @water_source = @data['holy_water']
+      @aspect_to_primer = @all_theurgy_data['aspect_to_primer']
+      @aspect = @immortal_to_aspect[@immortal]
+      @primer_number = @aspect_to_primer[@aspect]
+      @npc_name = @data['primer_npc']
+      @room = @data['favor_altars'][@immortal]['id']
+      @altar_noun = @data['favor_altars'][@immortal]['adjective']
+      @water_holder = @settings.water_holder
+    end
+
     EquipmentManager.new.empty_hands
 
-    settings = get_settings
+    if args.immortal.is_a?(String)
+      use_altar
+    else
+      use_puzzle
+    end
+  end
+
+  def use_altar
+    unless @water_holder
+      echo('You must purchase a water container and set water_holder: in your yaml to use altars for favors')
+      exit
+    end
+    if @primer_number > 300
+      ensure_copper_on_hand(750, @settings)
```

Comment:
💡 Is 750 copper so much that we can't just always grab that?

Reply:
nope, would you prefer that? I figured it might save a trip if someone uses 100-199 primers with a light aspect god with the default 3 silver on hand money.

Reply:
Eh, I don't like magic numbers, but it's not a big deal.